src/bridge/command.rs

```diff
@@ -32,23 +32,27 @@ fn set_windows_creation_flags(cmd: &mut TokioCommand) {
 
 fn build_nvim_cmd() -> TokioCommand {
     if let Some(path) = SETTINGS.get::<CmdLineSettings>().neovim_bin {
-        // if neovim_bin contains a path separator, then try to launch it directly
-        // otherwise use which to find the fully path
-        if path.contains('/') || path.contains('\\') {
-            if neovim_ok(&path) {
-                return build_nvim_cmd_with_args(&path);
-            }
-        } else if let Some(path) = platform_which(&path) {
-            if neovim_ok(&path) {
-                return build_nvim_cmd_with_args(&path);
+        if let Some(path) = shlex::split(&path) {
+            if let Some((bin, args)) = path.split_first() {
```

Comment:
No. `shlex::split` returns only when it failed to tokenize the input. So it can return an empty array when the input is a valid empty command line such as `""`. This check is necessary.

```rust
dbg!(shlex::split("'foo")); // -> None
dbg!(shlex::split("")); // -> Some([])

Reply:
I understood your intention. I'll try to reduce indentation. I also noticed I can remove one redundant clone so I'll remove it as well.